### PR TITLE
feat: handle passive and select constraints

### PR DESCRIPTION
**Pull request by LiKang6688**
There are three different constraints you can apply:

passive: turn off interactions like tooltips.
active: turn off interactions that affect the state of the visual representation like zoom, scroll, etc.
select: turn off selections.

```
{
  context: {
    constraints: {
      active: true,
      passive: true,
      select: true,
    },
  },
};
```

### CODE REVIEW

**Comment by haxxmaxx** [comment]
Has this changed? I thought active blocked everything?

**Comment by LiKang6688** [comment]
> Has this changed? I thought active blocked everything?

no, it was not changed. I got it from https://qlik.dev/libraries-and-tools/nebulajs/configuration#constraints and https://github.com/qlik-oss/nebula.js/blob/master/docs/embed-configuration.md#constraints

File: mashup-example/using-build-tool/qlik-sense-desktop/configure.js [hunk 1]
```diff
@@ -26,7 +26,9 @@ const nuked = embed.createConfiguration({
     language: 'en-US',
     keyboardNavigation: true,
     constraints: {
-      active: true, // do not allow interactions
+      active: false, // turn off interactions that affect the state of the visual representation like zoom, scroll, etc.
```
Comment (haxxmaxx) [inline mashup-example/using-build-tool/qlik-sense-desktop/configure.js:29]: change the comment

Comment (LiKang6688) [reply mashup-example/using-build-tool/qlik-sense-desktop/configure.js:29]: how about "turn off interactions that affect the state of the visual representation including selection, zoom, scroll, etc"

File: src/table/components/TableBodyWrapper.jsx [hunk 2]
```diff
@@ -25,7 +25,9 @@ function TableBodyWrapper({
   const hoverEffect = layout.components?.[0]?.content?.hoverEffect;
   const bodyStyle = useMemo(() => getBodyStyle(layout, theme), [layout, theme.name()]);
 
-  const selectionsEnabled = !!selectionsAPI && !constraints.active;
+  // active: turn off interactions that affect the state of the visual representation like zoom, scroll, etc.
+  // select: turn off selections.
+  const selectionsEnabled = !!selectionsAPI && !constraints.active && !constraints.select;
```
[the remark this answers is not in the document]
Comment (LiKang6688) [reply src/table/components/TableBodyWrapper.jsx:30]: I realized that checking active is not enough,
when active is false and select is true, selectionsEnabled should be false.

File: mashup-example/using-build-tool/qlik-sense-desktop/configure.js (hunk 1 above)
Comment (haxxmaxx) [reply mashup-example/using-build-tool/qlik-sense-desktop/configure.js:29]: sounds good

File: src/table/components/TableBodyWrapper.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (haxxmaxx) [reply src/table/components/TableBodyWrapper.jsx]: So only check select?

File: src/table/components/TableBodyWrapper.jsx [hunk 3]
```diff
@@ -25,6 +25,8 @@ function TableBodyWrapper({
   const hoverEffect = layout.components?.[0]?.content?.hoverEffect;
   const bodyStyle = useMemo(() => getBodyStyle(layout, theme), [layout, theme.name()]);
 
+  // active: turn off interactions that affect the state of the visual representation including selection, zoom, scroll, etc.
+  // select: turn off selections.
   const selectionsEnabled = !!selectionsAPI && !constraints.active;
```
Comment (haxxmaxx) [inline src/table/components/TableBodyWrapper.jsx:30]: Bit now you don't check constraints.select?

Comment (LiKang6688) [reply src/table/components/TableBodyWrapper.jsx]: wrong commit before. Fixed. Thanks.

**Review by haxxmaxx — APPROVED** [review]

**Pull request merged**